src/engine/sidechain/shoutoutput.cpp

```diff
@@ -0,0 +1,852 @@
+        } else if (m_pProfile->getEnabled()) {
+            m_lastErrorStr = shout_get_error(m_pShout);
+            qDebug() << "EngineBroadcast::processConnect() error:"
+                     << m_iShoutStatus << m_lastErrorStr;
+        }
+    }
+
+    // no connection, clean up
+    shout_close(m_pShout);
+    // delete m_encoder calls write() check if it will be exit early
+    DEBUG_ASSERT(m_iShoutStatus != SHOUTERR_CONNECTED);
+    m_encoder.reset();
+    if (m_pProfile->getEnabled()) {
+        m_pStatusCO->forceSet(STATUSCO_FAILURE);
+    } else {
+        m_pStatusCO->forceSet(STATUSCO_UNCONNECTED);
+    }
+    qDebug() << "EngineBroadcast::processConnect() returning false";
+    return false;
+}
+
+bool ShoutOutput::processDisconnect() {
+    qDebug() << "EngineBroadcast::processDisconnect()";
+    bool disconnected = false;
+    if (isConnected()) {
+        // We are connected but broadcast is disabled. Disconnect.
+        shout_close(m_pShout);
+        m_iShoutStatus = SHOUTERR_UNCONNECTED;
+        emit(broadcastDisconnected());
+        disconnected = true;
+    }
+    // delete m_encoder calls write() check if it will be exit early
+    DEBUG_ASSERT(m_iShoutStatus != SHOUTERR_CONNECTED);
+    m_encoder.reset();
+    return disconnected;
+
+}
+
+void ShoutOutput::write(const unsigned char *header, const unsigned char *body,
+                            int headerLen, int bodyLen) {
+    if (!m_pShout || m_iShoutStatus != SHOUTERR_CONNECTED) {
+        // This happens when the decoder calls flush() and the connection is
+        // already down
+        return;
+    }
+
+    // Send header if there is one
+    if (headerLen > 0) {
+        if(!writeSingle(header, headerLen)) {
+            return;
+        }
+    }
+
+    if(!writeSingle(body, bodyLen)) {
+        return;
+    }
+
+    ssize_t queuelen = shout_queuelen(m_pShout);
+    if (queuelen > 0) {
+        qDebug() << "shout_queuelen" << queuelen;
+        if (queuelen > kMaxNetworkCache) {
+            m_lastErrorStr = tr("Network cache overflow");
+            tryReconnect();
+        }
+    }
+}
+// These are not used for streaming, but the interface requires them
+int ShoutOutput::tell() {
+    if (!m_pShout) {
+        return -1;
+    }
+    return -1;
+}
+// These are not used for streaming, but the interface requires them
+void ShoutOutput::seek(int pos) {
+    Q_UNUSED(pos)
+    return;
+}
+// These are not used for streaming, but the interface requires them
+int ShoutOutput::filelen() {
+    return 0;
+}
+
+bool ShoutOutput::writeSingle(const unsigned char* data, size_t len) {
+    int ret = shout_send_raw(m_pShout, data, len);
+    if (ret == SHOUTERR_BUSY) {
+        // in case of busy, frames are queued
+        // try to flush queue after a short sleep
+        qDebug() << "EngineBroadcast::writeSingle() SHOUTERR_BUSY, trying again";
+        usleep(10000); // wait 10 ms until "busy" is over. TODO() tweak for an optimum.
```

Comment:
Does this actually happen? This assumes to be in a single thread without no other connections. 
I think we cannot use a common thread. We need a single thread for any connection, this way we can use the old way of operation as we like.